### PR TITLE
[UI Tests] - Add note to existing order test

### PR DESCRIPTION
**Pull request by jostnes**
## Description
A regression was fixed in https://github.com/woocommerce/woocommerce-ios/pull/10222 in which users couldn't add a note to an existing order. This PR adds a simple UI test to catch it from happening again, since this was a user-reported issue I think it's worth to be added as a UI test.

Originally, this was supposed to be a simpler change where I would expand the existing `test_load_existing_order()` test to include a `addCustomerNote()` step, however, found that due to the way we're using mocks (from the test, a new note is added but the mock file that is served for existing orders is still the file before the note is added -  we could start thinking about using stateful mocks but that's a separate topic for another time 🙂), that would end up with an error which I think can be more confusing than helpful (error not seen when not using mocks). See last second:

https://github.com/woocommerce/woocommerce-ios/assets/17252150/380a17a4-b37b-4a27-b8c7-46611b4157dc

Because of that, I decided to create a separate UI test to verify that the customer note section can be edited and we can catch this before our users do.

## Testing instructions
New test `test_add_note_to_existing_order` should work as expected. 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/UITestsFoundation/Screens/Orders/CustomerNoteScreen.swift [hunk 1]
```diff
@@ -31,6 +31,13 @@ public final class CustomerNoteScreen: ScreenObject {
         return self
     }
 
+    @discardableResult
+    public func verifyNoteAdded(_ text: String) throws -> Self {
+        XCTAssertTrue(app.keyboards.firstMatch.exists)
+        XCTAssertEqual(noteTextEditor.value as? String, text)
+        return self
+    }
```
Comment (pachlava) [inline WooCommerce/UITestsFoundation/Screens/Orders/CustomerNoteScreen.swift:39]: Just thinking aloud: this checks if customer note exists, but it does not know if it's because it was just added, or it was there for a long time. It can as well be called `verifyNoteExists` (but then, I suppose, keyboard should not be present 🤔).

After some thinking, though, I guess the combination of text + keyboard presence are signs of a just added note, so keeping it as is, is also fine.

File: WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift [hunk 2]
```diff
@@ -13,6 +13,8 @@ final class OrdersTests: XCTestCase {
         try LoginFlow.login()
     }
 
+    let customerNote = "New customer note"
+
```
Comment (pachlava) [inline WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift:17]: Knowing that this is only used in the new test, WDYT of moving it inside the test?

File: WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift [hunk 3]
```diff
@@ -54,6 +56,17 @@ final class OrdersTests: XCTestCase {
             .verifySingleOrderScreenLoaded()
     }
 
+    func test_add_note_to_existing_order() throws {
+        let orders = try GetMocks.readOrdersData()
+
+        try TabNavComponent().goToOrdersScreen()
+            .tapOrder(byOrderNumber: orders[0].number)
```
Comment (pachlava) [inline WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift:63]: This is not PR-related, but something that caught my eye. Using `number` word in this context might give a false impression that this is about tapping by index (or it's just me 🤔). Using `id` property of mocks would be more descriptive in this sense, but again, this is out of scope.

**Review by pachlava — APPROVED** [review]
Thank you, @jostnes! Local testing with iPhone and iPad went with no issues. I left a few notes, but they are more of reflection kind, not suggestions. :shipit:

File: WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift (hunk 2 above)
Comment (jostnes) [reply WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift:17]: so i'm thinking that we could define all `Strings` near the top of the screen so in the case that have to update them, it's easier to find near the top of the screen (though this is not standard across all test files yet)

File: WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift (hunk 3 above)
Comment (jostnes) [reply WooCommerce/WooCommerceUITests/Tests/OrdersTests.swift:63]: ah yes i could see how that could happen, since this is a property that is used across multiple tests, i'll update it in a follow-up PR (check if it's possible to use `id` instead of `number` and update for all tests)
